src/mscorlib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -957,12 +949,88 @@ internal static String Format(DateTime dateTime, String format, DateTimeFormatIn
             }
 
             if (format.Length == 1) {
+                switch (format[0]) {
+                    case 'o':
+                    case 'O':
+                        // Fast track for round trip format without going through a format string.
+                        return RoundTripFormat(ref dateTime);
+                }
+
                 format = ExpandPredefinedFormat(format, ref dateTime, ref dtfi, ref offset);
-            }            
+            }      
 
             return (FormatCustomized(dateTime, format, dtfi, offset));
         }
-    
+
+        internal static string RoundTripFormat(ref DateTime dateTime)
```

Comment:
>RoundTripFormat [](start = 31, length = 15)

one last thing, you named this method as RoundtripFormat which is same name as the defined string constant. would be better to rename the method to something better (e.g. FormatAsISO8601() or anything you think better).
internal const String RoundtripFormat = "yyyy'-'MM'-'dd'T'HH':'mm':'ss.fffffffK";